influxdb/src/main/scala/akka/stream/alpakka/influxdb/impl/InfluxDBFlowStage.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.influxdb.impl
+
+import akka.stream._
+import akka.stream.alpakka.influxdb.{InfluxDBWriteMessage, InfluxDBWriteResult}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import org.influxdb.InfluxDB
+import org.influxdb.impl.InfluxDBResultMapperHelper
+
+import scala.collection.immutable
+import org.influxdb.BatchOptions
+import org.influxdb.dto.{BatchPoints, Point}
+
+/**
+ * INTERNAL API
+ */
+private[influxdb] class InfluxDBFlowStage[T, C](
+    clazz: Option[Class[T]],
+    influxDB: InfluxDB
+) extends GraphStage[FlowShape[immutable.Seq[InfluxDBWriteMessage[T, C]], immutable.Seq[InfluxDBWriteResult[T, C]]]] {
+  private val in = Inlet[immutable.Seq[InfluxDBWriteMessage[T, C]]]("in")
+  private val out = Outlet[immutable.Seq[InfluxDBWriteResult[T, C]]]("out")
+
+  override val shape = FlowShape(in, out)
+
+  override protected def initialAttributes: Attributes =
+    super.initialAttributes and Attributes(ActorAttributes.IODispatcher)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    clazz match {
+      case Some(c) => new InfluxDBMapperRecordLogic
+      case None => new InfluxDBRecordLogic
+    }
+
+  sealed abstract class InfluxDBLogic extends GraphStageLogic(shape) with InHandler with OutHandler {
```

Comment:
I find it often easier to evolve the code when the logic is a non-inner class, as you do with `InfluxDBSourceLogic`.